query-coordinator/src/main/scala/com/socrata/querycoordinator/CompoundQueryRewriter.scala

```diff
@@ -0,0 +1,112 @@
+package com.socrata.querycoordinator
+
+import com.socrata.querycoordinator.QueryRewriter.{Anal, ColumnRef, Expr, RollupName}
+import com.socrata.soql.typed.ColumnRef
+import com.socrata.soql.{BinaryTree, Compound, Leaf, PipeQuery, SoQLAnalysis, typed}
+
+/**
+  * Rewrite compound query with either exact tree match or prefix tree match
+  * TODO: Make matching more flexible.  e.g. "SELECT a, b" does not match "SELECT b, a" or "SELECT a"
+  */
+trait CompoundQueryRewriter { this: QueryRewriter =>
+
+  /**
+    * match if the whole tree is equal or prefix is equal
+    */
+  def possibleRewrites(q: BinaryTree[Anal], rollups: Map[RollupName, BinaryTree[Anal]], debug: Boolean): (BinaryTree[Anal], Seq[String]) = {
+    if (!debug) {
```

Comment:
I might be missing something here but I don't understand why we're deactivating all this if debug is false? Presumably we want rollups on compound queries to work even if you're not making a debug query?

Reply:
We are ignoring the refresh problem.  This whole thing is just an experiment.  We are not allowing customers to use this feature yet.